### PR TITLE
[api] fix issue #3169

### PR DESCRIPTION
**Pull request by hero-maker**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by CoderZhi** [comment]
Please fill up the description of this PR

File: ioctl/newcmd/account/accountimport.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	importKeyUses, _ := client.SelectTranslation(importKeyCmdUses)
+	importKeyShorts, _ := client.SelectTranslation(importKeyCmdShorts)
+	accountImportKeyCmd := &cobra.Command{
+		Use:   importKeyUses,
+		Short: importKeyShorts,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			err := accountImportKey(client, cmd, args)
+			return output.PrintError(err)
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport.go:69]: `return accountImportKey(client, cmd, args)`
what's the value of this private function?

File: ioctl/newcmd/account/accountimport.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		},
+	}
+	return accountImportKeyCmd
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport.go:72]: `return &cobra.Command{...}`

File: ioctl/newcmd/account/accountimport.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+func NewAccountImportKeyStoreCmd(client ioctl.Client) *cobra.Command {
+	// accountImportKeyStoreCmd represents the account import keystore command
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport.go:76]: delete

File: ioctl/newcmd/account/accountimport.go [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+	accountImportCmd.AddCommand(NewAccountImportPemCmd(client))
+
+	return accountImportCmd
+}
+
+func validateAlias(alias string) error {
+	if err := validator.ValidateAlias(alias); err != nil {
+		return err
+	}
+	if addr, ok := config.ReadConfig.Aliases[alias]; ok {
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport.go:128]: use `client.XXX` instead

File: ioctl/newcmd/account/accountimport.go [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+		return fmt.Errorf("alias \"%s\" has already used for %s", alias, addr)
+	}
+	return nil
+}
+
+func writeToFile(alias, addr string) error {
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport.go:134]: use `client.XXX` function instead

File: ioctl/newcmd/account/accountimport.go [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+	return nil
+}
+
+func readPasswordFromStdin() (string, error) {
+	password, err := util.ReadSecretFromStdin()
+	if err != nil {
+		return "", fmt.Errorf("failed to get password")
+	}
+	return password, nil
+}
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport.go:155]: use `client.XXX` function instead

File: ioctl/newcmd/account/accountimport.go [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+func accountImportKey(client ioctl.Client, cmd *cobra.Command, args []string) error {
+	// Validate inputs
+	alias := args[0]
+	err := validateAlias(alias)
+	if err != nil {
+		return output.NewError(output.ValidationError, "invalid alias", err)
+	}
+	output.PrintQuery(fmt.Sprintf("#%s: Enter your private key, "+
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport.go:163]: use `client.XXX` function instead

File: ioctl/newcmd/account/accountimport_test.go [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+	defer ctrl.Finish()
+	client := mock_ioctlclient.NewMockClient(ctrl)
+	client.EXPECT().SelectTranslation(gomock.Any()).Return("", config.English).AnyTimes()
+	client.EXPECT().Config().Return(config.Config{}).AnyTimes()
+
+	cmd := NewAccountImportCmd(client)
+	result, err := util.ExecuteCmd(cmd, "hh")
+
+	require.NotNil(result)
+	require.Error(err)
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport_test.go:36]: what's the expected error?

File: ioctl/newcmd/account/accountimport_test.go [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+	defer ctrl.Finish()
+	client := mock_ioctlclient.NewMockClient(ctrl)
+	client.EXPECT().SelectTranslation(gomock.Any()).Return("", config.English).AnyTimes()
+	client.EXPECT().Config().Return(config.Config{}).AnyTimes()
+
+	cmd := NewAccountImportKeyCmd(client)
+	result, err := util.ExecuteCmd(cmd, "hh")
+
+	require.NotNil(result)
+	require.Error(err)
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport_test.go:51]: what's the expected error?

File: ioctl/newcmd/account/accountimport_test.go [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+	require.NoError(os.MkdirAll(testAccountFolder, os.ModePerm))
+	defer func() {
+		require.NoError(os.RemoveAll(testAccountFolder))
+	}()
+
+	cmd := NewAccountImportKeyStoreCmd(client)
+	result, err := util.ExecuteCmd(cmd, []string{"hh", "./testAccountImportKeyStoreCmd"}...)
+
+	require.NotNil(result)
+	require.Error(err)
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport_test.go:72]: what's the expected error?

File: ioctl/newcmd/account/accountimport_test.go [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+	defer ctrl.Finish()
+	client := mock_ioctlclient.NewMockClient(ctrl)
+	client.EXPECT().SelectTranslation(gomock.Any()).Return("", config.English).AnyTimes()
+	client.EXPECT().Config().Return(config.Config{}).AnyTimes()
+
+	cmd := NewAccountImportPemCmd(client)
+	result, err := util.ExecuteCmd(cmd, []string{"hh", "./testAccountImportPemCmd"}...)
+
+	require.NotNil(result)
+	require.Error(err)
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountimport_test.go:87]: what's the expected error?

**Review by CoderZhi — COMMENTED** [review]
@huof6829 please review details

File: ioctl/newcmd/account/accountimport.go [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+		"which will not be exposed on the screen.", alias))
+	privateKey, err := readPasswordFromStdin()
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:165]: use client.ReadSecret instead

File: ioctl/newcmd/account/accountimport.go [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+	}
+	return writeToFile(alias, addr)
+}
+
+func accountImportKeyStore(client ioctl.Client, cmd *cobra.Command, args []string) error {
+	// Validate inputs
+	alias := args[0]
+	err := validateAlias(alias)
+	if err != nil {
+		return output.NewError(output.ValidationError, "invalid alias", err)
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:182]: use errors.Wrap instead. output package is no longer use in newcmd

File: ioctl/newcmd/account/accountimport.go [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+	}
+	_, err = os.Stat(args[1])
+	if err != nil {
+		return output.NewError(output.ReadFileError, "", err)
+	}
+
+	output.PrintQuery(fmt.Sprintf("#%s: Enter your password of keystore, "+
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:189]: use  cmd.Println  instead

File: ioctl/newcmd/account/accountimport.go [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+		"which will not be exposed on the screen.", alias))
+	password, err := util.ReadSecretFromStdin()
+	if err != nil {
+		return output.NewError(output.InputError, "failed to get password", err)
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:193]: the same

File: ioctl/newcmd/account/accountimport.go [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+	}
+	addr, err := newAccountByKeyStore(client, cmd, alias, password, args[1])
+	if err != nil {
+		return output.NewError(0, "", err)
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:197]: the same

File: ioctl/newcmd/account/accountimport.go [hunk 17]
```diff
@@ -0,0 +1,225 @@
...
+	}
+	return writeToFile(alias, addr)
+}
+
+func accountImportPem(client ioctl.Client, cmd *cobra.Command, args []string) error {
+	// Validate inputs
+	alias := args[0]
+	err := validateAlias(alias)
+	if err != nil {
+		return output.NewError(output.ValidationError, "invalid alias", err)
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:207]: the same

File: ioctl/newcmd/account/accountimport.go [hunk 18]
```diff
@@ -0,0 +1,225 @@
...
+	}
+	_, err = os.Stat(args[1])
+	if err != nil {
+		return output.NewError(output.ReadFileError, "", err)
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:211]: the same

File: ioctl/newcmd/account/accountimport.go [hunk 19]
```diff
@@ -0,0 +1,225 @@
...
+	}
+
+	output.PrintQuery(fmt.Sprintf("#%s: Enter your password of pem file, "+
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:214]: use cmd.Println instead

File: ioctl/newcmd/account/accountimport.go [hunk 20]
```diff
@@ -0,0 +1,225 @@
...
+		"which will not be exposed on the screen.", alias))
+	password, err := util.ReadSecretFromStdin()
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:216]: use client.ReadSecret instead

File: ioctl/newcmd/account/accountimport.go [hunk 21]
```diff
@@ -0,0 +1,225 @@
...
+		config.English: "Import IoTeX key from pem file into wallet",
+		config.Chinese: "将IoTeX私钥从pem文件导入钱包",
+	}
+	importPemCmdUses = map[config.Language]string{
+		config.English: "pem ALIAS FILEPATH",
+		config.Chinese: "pem 别名 文件路径",
+	}
+)
+
+func NewAccountImportKeyCmd(client ioctl.Client) *cobra.Command {
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:59]: add comment for public function

File: ioctl/newcmd/account/accountimport.go [hunk 22]
```diff
@@ -0,0 +1,225 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			err := accountImportKey(client, cmd, args)
+			return output.PrintError(err)
+		},
+	}
+	return accountImportKeyCmd
+}
+
+func NewAccountImportKeyStoreCmd(client ioctl.Client) *cobra.Command {
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:75]: add comment

File: ioctl/newcmd/account/accountimport.go [hunk 23]
```diff
@@ -0,0 +1,225 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			err := accountImportKeyStore(client, cmd, args)
+			return output.PrintError(err)
+		},
+	}
+	return accountImportKeyStoreCmd
+}
+
+func NewAccountImportPemCmd(client ioctl.Client) *cobra.Command {
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:92]: add comment

File: ioctl/newcmd/account/accountimport.go [hunk 24]
```diff
@@ -0,0 +1,225 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			err := accountImportPem(client, cmd, args)
+			return output.PrintError(err)
+		},
+	}
+	return accountImportPemCmd
+}
+
+func NewAccountImportCmd(client ioctl.Client) *cobra.Command {
```
Comment (huof6829) [inline ioctl/newcmd/account/accountimport.go:109]: add comment

File: ioctl/newcmd/account/accountimport.go [hunk 25]
```diff
@@ -0,0 +1,225 @@
...
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package account
+
+import (
+	"fmt"
+	"github.com/iotexproject/iotex-core/ioctl"
+	"os"
+	"sigs.k8s.io/yaml"
```
Comment (dustinxie) [inline ioctl/newcmd/account/accountimport.go:13]: can use existing "gopkg.in/yaml.v2", let's not introduce a new dependency

**Comment by CoderZhi** [comment]
@huof6829 please take over this PR

**Comment by huof6829** [comment]
> @huof6829 please take over this PR

ok. i see

File: ioctl/newcmd/account/accountimport.go (hunk 7 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountimport.go]: done

File: ioctl/newcmd/account/accountimport.go (hunk 5 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountimport.go]: done

File: ioctl/newcmd/account/accountimport.go (hunk 4 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountimport.go]: done

File: ioctl/newcmd/account/accountimport.go (hunk 3 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountimport.go]: done

File: ioctl/newcmd/account/accountimport_test.go (hunk 8 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountimport_test.go]: added

File: ioctl/newcmd/account/accountimport_test.go (hunk 9 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountimport_test.go]: added

File: ioctl/newcmd/account/accountimport_test.go (hunk 10 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountimport_test.go]: added

File: ioctl/newcmd/account/accountimport.go (hunk 25 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountimport.go]: yes,   "sigs.k8s.io/yaml"  is deleted

File: ioctl/newcmd/account/accountimport.go [hunk 26]
```diff
@@ -0,0 +1,181 @@
...
+			return client.SetAlias(alias, addr)
+		},
+	}
+}
+
+func validateAlias(client ioctl.Client, alias string) error {
+	if err := validator.ValidateAlias(alias); err != nil {
+		return err
+	}
+	if addr, ok := client.Config().Aliases[alias]; ok {
```
[the remark this answers is not in the document]
Comment (huof6829) [reply ioctl/newcmd/account/accountimport.go:177]: no.  in AliasMap, key is accountAddr ,  value is aliasName

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**